validate-index.cpp

```diff
@@ -0,0 +1,92 @@
+#include "clang/Index/IndexDataStore.h"
+#include "clang/Index/IndexUnitReader.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/FileSystem.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+using namespace llvm;
+using namespace llvm::sys;
+using namespace clang::index;
+
+static cl::opt<std::string> IndexStore(cl::Positional, cl::Required,
+                                       cl::desc("<indexstore>"));
+
+int main(int argc, char **argv) {
+  cl::ParseCommandLineOptions(argc, argv);
+
+  std::string storeError{};
+  auto store = IndexDataStore::create(IndexStore, storeError);
+  if (not store) {
+    errs() << "error: failed to open indexstore " << IndexStore << " -- "
+           << storeError << "\n";
+    return EXIT_FAILURE;
+  }
+
+  std::vector<std::string> unitNames{};
+  store->foreachUnitName(false, [&](StringRef unitName) {
+    unitNames.push_back(unitName.str());
+    return true;
+  });
+
+  auto exitStatus = EXIT_SUCCESS;
+  for (const auto &unitName : unitNames) {
+    std::string readerError;
+    auto reader = IndexUnitReader::createWithUnitFilename(unitName, IndexStore,
+                                                          readerError);
+    if (not reader) {
+      errs() << "error: failed to read unit file " << unitName << " -- "
```

Comment:
what's the reasoning behind the usage of `errs()` and `outs()` below?

Reply:
I added a comment about this in the new `logMissingFile()` function. I want the output to be greppable. Also, being able to separate errors the tool hits vs errors the tool generates, seems useful. For example, since the tool's findings are printed to `outs()`, you can redirect the output to file, and still see these error messages in the console/log.